bin/pgenv

```diff
@@ -638,16 +674,27 @@ pgenv_configuration_write_variable(){
 # Writes the whole current configuration to the current version file
 # or the version file specified. It does a backup copy in the case
 # the file already exists.
+#
+# If the environment variable PGENV_CONFIGURATION_FILE is set
+# no configuration file (over)writing will happen: it is supposed
+# the user has declared the variable and thus it is not required to
+# update the configuration.
 pgenv_configuration_write() {
     local v=$1
     CONF=$( pgenv_configuration_file_name $v )
 
     # sanity check
     if [ -z "$CONF" ]; then
-        echo "Cannot determine which configuration file to use!"
+        echo "Cannot determine which configuration file to use!" > 2
         exit 1
     fi
 
+    # avoid overwriting configuration if the environment variable is set
+    if [ ! -z "${PGENV_CONFIGURATION_FILE}" ]; then
+	pgenv_debug "Cannot write configuration file while PGENV-CONFIGURATION_FILE set!"
+	exit 1
```

Comment:
This addition is causing the build script to error, signalling a failed build, when building by a `default.conf` present in the default `config` directory.

Debug output (removed the build output, but build succeeded)
```
citus@6c4d7721b906:~$ echo $PGENV_CONFIGURATION_FILE

citus@6c4d7721b906:~$ PGENV_DEBUG=1 pgenv build 14.9
[DEBUG] Looking for configuration in /home/citus/.pgenv/config/14.9.conf
[DEBUG] Looking for configuration in /home/citus/.pgenv/config/default.conf
[DEBUG] Load configuration from [/home/citus/.pgenv/config/default.conf]
[DEBUG] Downloading tarball
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 21.1M  100 21.1M    0     0  3751k      0  0:00:05  0:00:05 --:--:-- 3803k
[DEBUG] Patching version 14.9
[DEBUG] Patch index file []
[DEBUG] Skipping patch: no index file found!
[DEBUG] configure command line [--prefix=/home/citus/.pgenv/pgsql-14.9] + [--enable-debug --enable-depend --enable-cassert --enable-tap-tests CFLAGS=-ggdb -Og -g3 -fno-omit-frame-pointer -DUSE_VALGRIND --with-openssl --with-libxml --with-libxslt --with-uuid=e2fs --with-icu --with-lz4]

...

[DEBUG] Configuration file overwritten by environment variable PGENV_CONFIGURATION_FILE = /home/citus/.pgenv/config/default.conf
[DEBUG] Cannot write configuration file while PGENV-CONFIGURATION_FILE set!
citus@6c4d7721b906:~$ echo $?
1
citus@6c4d7721b906:~$ echo $PGENV_CONFIGURATION_FILE

citus@6c4d7721b906:~$
```

exit with a non-zero exit here is probably not desirable.

Example how this breaks automated workflows: https://github.com/citusdata/citus/actions/runs/6082295223/job/16499766357

It took me a while to understand this was caused by a change in pgenv instead of actual build failures, but running the steps of the build manually pointed me to this change.

Any idea how this should be solved? As you can see from the debug output `pgenv` assumes the `PGENV_CONFIGURATION_FILE` is set, although it is not, probably due to the config being resolved.

I believe the previous behaviour was to write the applied config of the build to a version specific file when seeded from a default config file. This is sane, as it makes rebuilds more deterministic.

minor nit; the log line has a small typo in there


```suggestion
	pgenv_debug "Cannot write configuration file while PGENV_CONFIGURATION_FILE set!"
	exit 1
```

Reply:
Checking the github actions link in an incognito window seems to indicate that non-contributors can't see the output of the `Build and push` step, but it fails on `exit 1` on `pgenv build <version>`.

The output pasted above is easier to digest as it has the big wall of text removed and actually contains the debug lines, cause is the same.